crates/rover-client/src/query/schema/stash.rs

```diff
@@ -0,0 +1,78 @@
+use crate::blocking::Client;
+use crate::RoverClientError;
+use graphql_client::*;
+
+#[derive(GraphQLQuery)]
+// The paths are relative to the directory where your `Cargo.toml` is located.
+// Both json and the GraphQL schema language are supported as sources for the schema
+#[graphql(
+    query_path = "src/query/schema/stash.graphql",
+    schema_path = "schema.graphql",
+    response_derives = "PartialEq, Debug, Serialize, Deserialize",
+    deprecated = "warn"
+)]
+/// This struct is used to generate the module containing `Variables` and
+/// `ResponseData` structs.
+/// Snake case of this name is the mod name. i.e. stash_schema_query
+pub struct StashSchemaMutation;
+
+/// Returns a message from apollo studio about the status of the update, and
+/// a sha256 hash of the schema to be used with `schema publish`
+pub fn run(
+    variables: stash_schema_mutation::Variables,
+    client: Client,
+) -> Result<(String, String), RoverClientError> {
```

Comment:
> this could also just be a `String` if it's easier, though I like to keep non-String types around if at all possible

Can you explain this or provide a link so I can read up? I'm not sure why that'd be beneficial

Reply:
I don't have a link really, this is more of just a personal preference where if there is a more specific data type you can provide, it makes it easier to extend/modify behavior moving forward in addition to making it obvious what TYPE of hash it actually is (instead of a String which could be any type of hash).

You'd just call `.to_string()` where you would want to use it instead of calling `.to_string()` before storing it in the struct.